pkg/util/validation/validate.go

```diff
@@ -216,6 +230,23 @@ func ValidateLabels(cfg LabelValidationConfig, userID string, ls []cortexpb.Labe
 		}
 
 		lastLabelName = l.Name
+		seriesLabelSizeBytes += l.Size()
+	}
+
+	if maxSeriesLabelSizeBytes > 0 && seriesLabelSizeBytes > maxSeriesLabelSizeBytes {
+		DiscardedSamples.WithLabelValues(seriesLabelSizeBytesExceeded, userID).Inc()
+		return newSeriesLabelSizeBytesExceededError(ls, seriesLabelSizeBytes, maxSeriesLabelSizeBytes)
+	}
+
+	maxSeriesLabelSizeBytesMutex.Lock()
+	defer maxSeriesLabelSizeBytesMutex.Unlock()
+	if prev, ok := maxSeriesLabelSizeBytesMap[userID]; ok {
+		if prev > maxSeriesLabelSizeBytes {
+			maxSeriesLabelSizeBytesMap[userID] = maxSeriesLabelSizeBytes
+			MaxSeriesLabelSizeBytesPerUser.WithLabelValues(userID).Set(float64(maxSeriesLabelSizeBytes))
+		}
+	} else {
+		maxSeriesLabelSizeBytesMap[userID] = maxSeriesLabelSizeBytes
```

Comment:
Im worried about this lock here!

Could we do this not on the hot path? Maybe use a defer or even do it on compactor?

Reply:
What would be the use case for these metrics?

Reply:
Before we can enable this, we need to make sure existing users are not impacted. This metrics provide a way to monitor the current largest series in terms of total bytes that each user have.

Reply:
I've removed this metric for now. The plan is to add it later in the private repo and not upstream.